manifests/config.pp

```diff
@@ -34,139 +34,160 @@
-
-  file { "${jira::webappdir}/bin/user.sh":
-    content => epp("${module_name}/user.sh.epp"),
-    mode    => '0755',
+  $validation_query_timeout = if $jira::validation_query_timeout {
+    $jira::validation_query_timeout
+  } else {
+    $jira::db ? {
+      'mysql' => 3,
+      default => undef,
+    }
   }
-
-  file { "${jira::webappdir}/bin/setenv.sh":
-    content => epp("${module_name}/setenv.sh.epp"),
-    mode    => '0755',
+  # JIRA will complain if these aren't set for PostgreSQL (will work fine though)
+  # https://confluence.atlassian.com/jirakb/connection-problems-to-postgresql-result-in-stuck-threads-in-jira-1047534091.html
+  $connection_settings = if $jira::connection_settings {
+    $jira::connection_settings
+  } else {
+    $jira::db ? {
+      'postgresql' => 'tcpKeepAlive=true;socketTimeout=240',
+      default => undef,
+    }
   }
-
-  $dbconfig_template = $jira::use_jndi_ds ? {
-    true    => "${module_name}/dbconfig.jndi.xml.epp",
-    default => "${module_name}/dbconfig.xml.epp"
+  $dbconfig_params = {
+    version => $jira::version,
+    is_installed => Deferred('jira::is_installed', [$jira::homedir]),
+    change_dbpassword => $jira::change_dbpassword,
+    dbtype => $dbtype,
+    dbschema => $dbschema,
+    dburl => pick($jira::dburl, $jira::db ? {
+        'postgresql' => "jdbc:${jira::db}://${jira::dbserver}:${dbport}/${jira::dbname}",
+        'mysql'      => "jdbc:${jira::db}://${jira::dbserver}:${dbport}/${jira::dbname}?useUnicode=true&amp;characterEncoding=UTF8&amp;sessionVariables=default_storage_engine=InnoDB",
+        'oracle'     => "jdbc:${jira::db}:thin:@${jira::dbserver}:${dbport}${oracle_separator}${jira::dbname}",
+        'sqlserver'  => "jdbc:jtds:${jira::db}://${jira::dbserver}:${dbport}/${jira::dbname}",
+        'h2'         => "jdbc:h2:file:/${jira::homedir}/database/${jira::dbname}",
+    }),
+    dbdriver => $dbdriver,
+    dbuser => $jira::dbuser,
+    dbpassword => $jira::dbpassword,
+    pool_min_size => pick($jira::pool_min_size, 20),
+    pool_max_size => $pool_max_size,
+    pool_max_wait => pick($jira::pool_max_wait, 30000),
+    pool_max_idle => $pool_max_idle,
+    pool_remove_abandoned => pick($jira::pool_remove_abandoned, true),
+    pool_remove_abandoned_timeout => pick($jira::pool_remove_abandoned_timeout, 300),
+    pool_test_on_borrow => pick($jira::pool_test_on_borrow, false),
+    pool_test_while_idle => pick($jira::pool_test_while_idle, true),
+    validation_query => $validation_query,
+    validation_query_timeout => $validation_query_timeout,
+    min_evictable_idle_time => pick($jira::min_evictable_idle_time, 60000),
+    time_between_eviction_runs => pick($jira::time_between_eviction_runs, 300000),
+    connection_settings => $connection_settings,
+    jndi_ds_name => $jira::jndi_ds_name,
   }
   file { "${jira::homedir}/dbconfig.xml":
-    content => epp($dbconfig_template),
+    content => Deferred('epp', [$dbconfig_template, $dbconfig_params]),
```

Comment:
Deferring the epp function failed.
Seems like puppet agent tries to open the template file on the agent side ...
```
root@test-role-jira:~> puppet agent -t
Info: Using environment 'vagrant'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Loading facts
Notice: Requesting catalog from puppet.vagrant.spk.no:8140 (192.168.200.2)
Notice: Catalog compiled by test-role-puppetmaster.spk.no
Info: Caching catalog for test-role-jira.spk.no
Info: Applying configuration version '1743501253'
Error: Failed to apply catalog: Could not find template 'jira/dbconfig.xml.epp'
```

Reply:
Solved by using inline_epp